apps/src/code-studio/progressRedux.ts

```diff
@@ -384,12 +388,43 @@ export const sendPredictLevelReport = createAsyncThunk<
     state: RootState;
   }
 >('progress/sendPredictLevelReport', async (payload, thunkAPI) => {
+  const extraPayload = {
+    program: payload.predictResponse,
+  };
   sendReportHelper(
     payload.appType,
     TestResults.CONTAINED_LEVEL_RESULT,
     thunkAPI.dispatch,
     thunkAPI.getState,
-    payload.predictResponse
+    extraPayload
+  );
+});
+
+export const sendSubmitReport = createAsyncThunk<
+  void,
+  {appType: string; submitted: boolean},
+  {
+    dispatch: AppDispatch;
+    state: RootState;
+  }
+>('progress/sendSubmitReport', async (payload, thunkAPI) => {
+  const extraPayload = {
+    submitted: payload.submitted.toString(),
+  };
+  const result = payload.submitted
+    ? TestResults.SUBMITTED_RESULT
+    : TestResults.UNSUBMITTED_ATTEMPT;
+  await sendReportHelper(
+    payload.appType,
+    result,
+    thunkAPI.dispatch,
+    thunkAPI.getState,
+    extraPayload
+  );
+  // Submit status isn't properly updated by just saving the status code, so re-query
+  // user progress to force the bubble to update.
+  thunkAPI.dispatch(
+    queryUserProgress(thunkAPI.getState().currentUser.userId.toString())
```

Comment:
It's a bummer we have to requery this (vs. optimistic updates or something), but it looks like it's a common pattern.

Reply:
Yeah in theory it should be possible, but the progress store is so complicated I couldn't figure out how to get the data plumbed through correctly. I justified this because a user should press submit rarely, probably only one time per submit level.